src/dht_proxy_server.cpp

```diff
@@ -201,164 +177,176 @@ DhtProxyServer::updateStats() const
     stats_.nodeInfo = nodeInfo_;
 }
 
-void
-DhtProxyServer::getNodeInfo(const Sp<restbed::Session>& session) const
+template <typename HttpResponse>
+HttpResponse DhtProxyServer::initHttpResponse(HttpResponse response) const
 {
-    requestNum_++;
-    const auto request = session->get_request();
-    int content_length = std::stoi(request->get_header("Content-Length", "0"));
-    session->fetch(content_length,
-        [this](const Sp<restbed::Session>& s, const restbed::Bytes& /*b*/) mutable
-        {
-            try {
-                if (dht_) {
-                    Json::Value result;
-                    {
-                        std::lock_guard<std::mutex> lck(statsMutex_);
-                        if (nodeInfo_.ipv4.good_nodes == 0 && nodeInfo_.ipv6.good_nodes == 0) {
-                            // NOTE: we want to avoid the disconnected state as much as possible
-                            // So, if the node is disconnected, we should force the update of the cache
-                            // and reconnect as soon as possible
-                            // This should not happen much
-                            nodeInfo_ = dht_->getNodeInfo();
-                        }
-                        result = nodeInfo_.toJson();
-                    }
-                    result["public_ip"] = s->get_origin(); // [ipv6:ipv4]:port or ipv4:port
-                    auto output = Json::writeString(jsonBuilder_, result) + "\n";
-                    s->close(restbed::OK, output);
-                }
-                else
-                    s->close(restbed::SERVICE_UNAVAILABLE, "{\"err\":\"Incorrect DhtRunner\"}");
-            } catch (...) {
-                s->close(restbed::INTERNAL_SERVER_ERROR, "{\"err\":\"Internal server error\"}");
-            }
-        }
-    );
+    response.append_header("Server", "RESTinio");
+    response.append_header(restinio::http_field::content_type, "application/json");
+    response.append_header(restinio::http_field::access_control_allow_origin, "*");
+    response.connection_keep_alive();
+    return response;
 }
 
-void
-DhtProxyServer::getStats(const Sp<restbed::Session>& session) const
+std::unique_ptr<RestRouter>
+DhtProxyServer::createRestRouter()
+{
+    using namespace std::placeholders;
+    auto router = std::make_unique<RestRouter>();
+    router->http_get("/", std::bind(&DhtProxyServer::getNodeInfo, this, _1, _2));
+    router->http_get("/stats", std::bind(&DhtProxyServer::getStats, this, _1, _2));
+    router->http_get("/:hash", std::bind(&DhtProxyServer::get, this, _1, _2));
+    router->http_post("/:hash", std::bind(&DhtProxyServer::put, this, _1, _2));
+    router->http_get("/:hash/listen", std::bind(&DhtProxyServer::listen, this, _1, _2));
```

Comment:
Not backward compatible. This will break all clients

Reply:
Yes, awaiting https://github.com/Stiffstream/restinio/issues/26 or I will PR soon

Reply:
Done at f8d62dc